apps/ledger-live-mobile/src/components/RequiresBLE/hooks/useRequireBluetooth.ts

```diff
@@ -0,0 +1,175 @@
+    isHookEnabled &&
+    Platform.OS === "android" &&
+    requiredFor === "scanning" &&
+    androidHasBluetoothPermissions === "granted" &&
+    bluetoothServicesState === "enabled";
+
+  const {
+    hasPermission: androidHasLocationPermission,
+    neverAskAgain: androidLocationPermissionNeverAskAgain,
+    requestForPermissionAgain:
+      androidLocationPermissionRequestForPermissionsAgain,
+  } = useAndroidLocationPermission({
+    isHookEnabled: isAndroidLocationPermissionHookEnabled,
+  });
+
+  // Handles location services on Android if necessary and once bluetooth permisions are granted and bluetooth services are enabled
+  // and location permission is granted
+  const isAndroidEnableLocationHookEnabled =
+    isHookEnabled &&
+    Platform.OS === "android" &&
+    requiredFor === "scanning" &&
+    androidHasBluetoothPermissions === "granted" &&
+    bluetoothServicesState === "enabled" &&
+    androidHasLocationPermission === "granted";
+
+  const {
+    locationServicesState,
+    checkAndRequestAgain: androidEnableLocationCheckAndRequestAgain,
+  } = useAndroidEnableLocation({
+    isHookEnabled: isAndroidEnableLocationHookEnabled,
+  });
+
+  let bluetoothRequirementsState: BluetoothRequirementsState = "unknown";
+  let retryRequestOnIssue = null;
+  let cannotRetryRequest = false;
+
+  let someUnknown = false;
+
+  // Logic to determine the state of the bluetooth requirements, and the action to perform to retry
+  // Because we check if each hook is enabled, we don't need to respect a specific order for the checks
+  if (isAndroidBluetoothPermissionsHookEnabled) {
+    if (androidHasBluetoothPermissions === "denied") {
+      bluetoothRequirementsState = "bluetooth_permissions_ungranted";
+      retryRequestOnIssue =
+        androidBluetoothPermissionsRequestForPermissionsAgain;
+      cannotRetryRequest = androidBluetoothPermissionsNeverAskAgain;
+    } else if (androidHasBluetoothPermissions !== "granted") {
+      someUnknown = true;
+    }
+  }
+
+  if (isAndroidLocationPermissionHookEnabled) {
+    if (androidHasLocationPermission === "denied") {
+      bluetoothRequirementsState = "location_permission_ungranted";
+      retryRequestOnIssue = androidLocationPermissionRequestForPermissionsAgain;
+      cannotRetryRequest = androidLocationPermissionNeverAskAgain;
+    } else if (androidHasLocationPermission !== "granted") {
+      someUnknown = true;
+    }
+  }
+
+  if (isEnableBluetoothHookEnabled) {
+    if (bluetoothServicesState === "disabled") {
+      bluetoothRequirementsState = "bluetooth_disabled";
+      retryRequestOnIssue = enableBluetoothCheckAndRequestAgain;
+    }
+    // Happens on iOS when user did not grant bluetooth permissions
+    // No retry function on iOS because there is currently no way to request bluetooth permissions directly
+    else if (bluetoothServicesState === "unauthorized") {
+      bluetoothRequirementsState = "bluetooth_permissions_ungranted";
```

Comment:
I know that `RequiresBluetoothDrawer` default `cannotRetryRequest` to `true` but shouldn't we still set it to true here? In case someone decides to consume this hook somewhere else.

Reply:
I agree, adding it !